tests/parser/types/test_node_types.py

```diff
@@ -0,0 +1,90 @@
+from pytest import raises
+
+from vyper.types import (
+    BaseType,
+    ByteArrayType,
+    ListType,
+    MappingType,
+    NullType,
+    StructType,
+    TupleType,
+    canonicalize_type,
+    get_size_of_type,
+)
+
+
+def test_null_type():
+    node1 = NullType()
```

Comment:
I'd prefer if this stuff was done in Vyper code instead of leveraging internal types

Reply:
Most of these scenarios test the Type classes directly - and couldn't be reached by writing vyper code. See this as a type of compiler-writer future proofing.